### DISCUSSION TITLE
Debug/trace mode

### DISCUSSION THREAD

**Discussion by scotttrinh** [pmndrs/use-gesture#514] [Ideas]
I'm working on adding tests to a component that uses multiple gestures and apart from the difficulties presented by jsdom's lack of `PointerEvent`, I'm also at a loss for how to troubleshoot why my pinching gesture, which I cribbed directly from the pinch tests here, are not triggering `onPinch*` events in the tests. The gestures fire in manual QA. The rest of the gestures are working in the tests as well, so there must be something about the events I'm firing that are not triggering the event. From a debugging perspective, there is no hook into the underlying library to see what events are being handled and why they aren't triggering certain gestures.

My idea is to provide a way to turn on additional trace logging so that troubleshooting why gestures aren't firing is a little easier.

**Comment by dbismut** [comment]
@scotttrinh happy to hear your thoughts about how to do this!
